enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -213,158 +163,154 @@ the underlying, associated Pod references CSI volumes using `Shares`
-- and that `DaemonSet` will have privileged containers.  See the
-  [csi-node-driver-registrar](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.17/hostpath/csi-hostpath-plugin.yaml#L48-L52)
-  container and the main [csi plugin
-  container](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.17/hostpath/csi-hostpath-plugin.yaml#L82-L83)
-  for the hostpath example and [secrets store
-  container](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/deploy/secrets-store-csi-driver.yaml#L61)
-  in the secrets store example.
-- When provisioning volumes, the plugin can access the node/host
-  filesystem, like for example the directory, corresponding to the
-  Pod's mount point, where the secret contents will be stored (though
-  we have some choices on exactly which OS level file system features
-  we employ to store the content ... see the Open Questions)
-- All the reference implementations reviewed to date leverage https://github.com/kubernetes/utils/blob/master/mount/mount.go
+- the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) details the CSI architecture, concepts, and protocol.
+
+#### Ephemeral CSI driver implementation
+
+The CSI driver will be deployed as a DaemonSet so that shared Secrets/ConfigMaps can be added to pods running on the node.
+Pods for this DaemonSet will need to run as privileged to access host paths on the node.
+The CSI driver will only need to support "ephemeral" volumes that are specified by the `csi:` volume type on Pods.
+There is no requirement for this CSI driver to provision persistent volumes.
 
 Some details around inline ephemeral volumes:
 - The employ a subset of the CSI specification flows, specifically only the `NodePublish` and `NodeUnpublish` CSI steps
-- the underlying k8s api object `CSIVolumeSource` has the same `volumeAttributes` for storing metadata to be passed to the CSI driver
-- there is only a single `LocalObjectReference` for a `Secret` reference in the same namespace as the consuming pod.  So that aspect does not help us, and will not factor into the solution.
-- The `volumeAttributes` field for metadata gives us enough capabilities for expressing both the name, namespace, and type of sharable API object to be consumed.  And who is requesting access.
+- The underlying k8s api object `CSIVolumeSource` has the same `volumeAttributes` for storing metadata to be passed to the CSI driver
+- There is only a single `LocalObjectReference` for a `Secret` reference in the same namespace as the consuming pod.
+  So that aspect does not help us, and will not factor into the solution.
+- The `volumeAttributes` field for metadata gives us enough capabilities for expressing both the name, namespace, and type of sharable API object to be consumed.
+  Using pod info on mount lets the driver see which service account is requesting access.
```

Comment:
this also provides for us
- pod name
- pod namespace 
- pod uid
- the fact we are using ephemeral